projects/sandbox/analyze/analyze.py

```diff
@@ -183,111 +196,65 @@ def build_background(
                     t,
-                    kernel_length=1.0,
+                    kernel_length=kernel_length,
                     window_length=window_length,
                     normalizer=normalizer,
                 )
-                future.add_done_callback(
-                    lambda f: pbar.update(analyze_task_id, advance=1)
+
+                pbar.update(analyze_task_id, advance=1)
+
+                # fit background to integrated output
+                # in the main process
+                background.fit((integrated, t), warm_start=warm_start)
+
+                # submit the writing job to our thread pool and
+                # use a callback to keep track of all the filenames
+                # for a given normalization window
+                shift_dir = get_write_dir(
+                    write_dir, shift, "background-integrated", norm
                 )
-                integration_futures[(norm, shift)] = [future]
 
-            # advance the task keeping track of how many files
-            # we've loaded by one
-            pbar.update(load_task_id, advance=1)
+                future = thread_ex.submit(
+                    write_timeseries,
+                    shift_dir,
+                    prefix="integrated",
+                    t=t,
+                    y=y,
+                    integrated=integrated,
+                )
 
-        # make sure we have the expected number of jobs submitted
-        if len(integration_futures) < (len(norm_seconds) * len(load_futures)):
-            raise ValueError(
-                "Expected {} integration jobs submitted, "
-                "but only found {}".format(
-                    len(norm_seconds) * len(load_futures),
-                    len(integration_futures),
+                future.add_done_callback(
+                    lambda f: pbar.update(write_task_id, advance=1)
                 )
-            )
 
-        # as the integration jobs come back, write their
-        # results using our thread pool and record the
-        # min and max values for our discrete distribution
-        segment_futures = []
-        for (norm, shift), (t, y, integrated) in as_completed(
-            integration_futures
-        ):
-            # submit the writing job to our thread pool and
-            # use a callback to keep track of all the filenames
-            # for a given normalization window
-            shift_dir = get_write_dir(
-                write_dir, shift, "background-integrated", norm
-            )
-            future = thread_ex.submit(
-                write_timeseries,
-                shift_dir,
-                t=t,
-                y=y,
-                integrated=integrated,
-            )
-            future.add_done_callback(
-                lambda f: pbar.update(write_task_id, advance=1)
-            )
-            fname_futures[norm].append(future)
-            segment_futures.append(future)
+                write_futures.append(future)
```

Comment:
I think you'll want to include a `wait` for these write futures at the end of this function yeah? Unless we get some speed improvement from starting to move to characterization while the last events finish writing.